### ISSUE TITLE
Error during runtime on the getter process with a Java reserved word

### ISSUE THREAD

**Issue by 988554114** [graphql-java-generator/graphql-maven-plugin-project#166]
Hello,

In the graphql specification schema we have some field like 'package, class...') and these fields are reserved words used by java, 
but we are on client part and normally .
We can't change schema.

The problem is when the graphql-java-generator parse these fields.
Client graphQL transform some attributs found into schema, e.g 'package' by the java field '_package' (because it is a java word)
and create a getPackage() method to acceded on _package field value.

But during the graphql client runtime, a get_package error is occured, because the get is build dynamically 
by fields introspection and the get_package is call and not found !.
I think that the problem can be fix by generate the field package_ instead _package.

maven version:
<dependency>
	<groupId>com.graphql-java-generator</groupId>
	<artifactId>graphql-java-client-runtime</artifactId>
	<version>1.18.7</version>
</dependency>

**Comment by etienne-sf** [comment]
Thanks for the report and the analysis.
These case would also trigger another issu in server's wiring. 


The correction will be available in next version, in some days.

Etienne

**Comment by etienne-sf** [comment]
Released in the 1.18.9.

**Comment by kewinremy** [comment]
Hi, although I am using the version 1.18.9, I still get this error.

```sh
	java.lang.RuntimeException: The com.example.gitlab.ComplianceFrameworkInput class should have a getter 'is_default', but this getter is missing
	at com.graphql_java_generator.util.GraphqlUtils.getGetter(GraphqlUtils.java:666)
	at com.graphql_java_generator.util.GraphqlUtils.invokeGetter(GraphqlUtils.java:688)
	at com.graphql_java_generator.client.request.InputParameter.getStringContentForAnInputTypeValue(InputParameter.java:871)
	at com.graphql_java_generator.client.request.InputParameter.getStringContentForGraphqlQuery(InputParameter.java:791)
	at com.graphql_java_generator.client.request.InputParameter.getStringContentForAnInputTypeValue(InputParameter.java:880)
	at com.graphql_java_generator.client.request.InputParameter.getStringContentForGraphqlQuery(InputParameter.java:791)
	at com.graphql_java_generator.client.request.InputParameter.getStringContentForGraphqlQuery(InputParameter.java:733)
	at com.graphql_java_generator.client.request.InputParameter.appendInputParametersToGraphQLRequests(InputParameter.java:936)
	at com.graphql_java_generator.client.request.QueryField.appendToGraphQLRequests(QueryField.java:346)
	at com.graphql_java_generator.client.request.QueryField.appendToGraphQLRequests(QueryField.java:376)
	at com.graphql_java_generator.client.request.AbstractGraphQLRequest.getPayload(AbstractGraphQLRequest.java:663)
	at com.graphql_java_generator.client.request.AbstractGraphQLRequest.buildRequestAsMap(AbstractGraphQLRequest.java:703)
	at com.graphql_java_generator.client.RequestExecutionSpringReactiveImpl.execute(RequestExecutionSpringReactiveImpl.java:162)
	at com.example.gitlab.util.MutationExecutor.createComplianceFrameworkWithBindValues(MutationExecutor.java:748)
	at com.example.gitlab.util.MutationExecutor.createComplianceFramework(MutationExecutor.java:692)
	at com.example.demo.Service.createComplianceFramework(Service.java:75)
	at com.example.demo.DemoApplication.main(DemoApplication.java:20)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.springframework.boot.devtools.restart.RestartLauncher.run(RestartLauncher.java:49)
Caused by: java.lang.NoSuchMethodException: com.example.gitlab.ComplianceFrameworkInput.is_default()
	at java.base/java.lang.Class.getMethod(Class.java:2227)
	at com.graphql_java_generator.util.GraphqlUtils.getGetter(GraphqlUtils.java:653)
	... 21 more
```

here is the class with its properties generated:
As you can see, the **default** reserved keyword has an underscore in the private field.

```
	@JsonProperty("default")
	@GraphQLScalar(fieldName = "default", graphQLTypeSimpleName = "Boolean", javaClass = java.lang.Boolean.class)
	java.lang.Boolean _default;


	public void setDefault(java.lang.Boolean _default) {
		this._default = _default;
	}

	public java.lang.Boolean getDefault() {
		return _default;
	}
```

The problematic class is this one.

```
	public Method getGetter(Class<?> clazz, Field field) {
		String getterMethodName = "get" + getPascalCase(field.getName());
		try {
			Method method = null;
			try {
				method = clazz.getMethod(getterMethodName);
			} catch (NoSuchMethodException e) {
				// For the boolean fields, the getter may be named isProperty. Let's try that:
				if (field.getType().equals(boolean.class) || field.getType().equals(Boolean.class)) {
					getterMethodName = "is" + getPascalCase(field.getName());
					method = clazz.getMethod(getterMethodName);
				} else {
					throw e;
				}
			}
```

According to the internal logic of the plugin for booleans, I extended the method in a PR.

```
	public Method getGetter(Class<?> clazz, Field field) {
		String getterMethodName = "get" + getPascalCase(field.getName());
		try {
			Method method = null;
			try {
				method = clazz.getMethod(getterMethodName);
			} catch (NoSuchMethodException e) {
				// For the boolean fields, the getter may be named isProperty. Let's try that:
				if (field.getType().equals(boolean.class) || field.getType().equals(Boolean.class) && field.getName()
						.startsWith("is")) {
					getterMethodName = "is" + getPascalCase(field.getName());
					method = clazz.getMethod(getterMethodName);
				}
				// For the fields conflicting with java keywords and generated with an underscore (_), the getter may be named _Property. Let's try that:
				else if (field.getName().startsWith("_")) {
					String sanitizedfield = field.getName().substring(1);
					getterMethodName = "get" + getPascalCase(sanitizedfield);
					method = clazz.getMethod(getterMethodName);
				} else {
					throw e;
				}
			}
			// The return type must be the same as the field's class
			if (field.getType() != method.getReturnType()) {
				throw new RuntimeException("The getter '" + getterMethodName + "' and the field '" + field.getName()
						+ "' of the class " + clazz.getName() + " should be of the same type");
			}
```

Please find below the PR:
